openstates/get-state-lawmakers.js

```diff
@@ -187,28 +135,77 @@ async function getStateLegs() {
-          const member = lawmakers[memberId];
-          member.thp_id = memberId;
-          checkIsInDb(member)
-            .then((isInDatabase) => {
-              if (isInDatabase) {
-                console.log('already there, updating', member.thp_id)
-                update(member);
+          const person = lawmakers[memberId]
+          const newOfficePerson = new StateLawmaker(memberId, person.openStatesDisplayName, person.state)
+
+          // Unpack open states data
+          newOfficePerson.unpackOpenStatesLawmaker(person);
+
+          // Handle person unpacking and storage based off if member already exists in database
+          newOfficePerson.checkDatabaseShortInfo()
+            .then((checkResult) => {
+              // If an object was returned, we either need to do to nothing, or update the existing data
+              if (checkResult) {
+                // Check the id of the person found matches the open states id
+                if (newOfficePerson.id == checkResult.id) {
+                    // Great, don't do anything, this data is already correct
+                    console.log(`office person: ${newOfficePerson.id} already exists`);
+                } else {
+                    // This member was added to the database prior to open states adding them
+                    // Get their full data, merge the already stored data with the open states data, and then update
+                    // the database with their official id and propagate changes back to the state legislature lookup
+                    console.log(`office person found by name; oldId: ${checkResult.id}, newId: ${newOfficePerson.id}`);
+
+                    // Get full office person data
+                    let existingOfficePersonRef = firebase.firestore.collection('office_people').doc(checkResult.id);
+                    let existingData = existingOfficePersonRef.get().then(doc => {
+                        // Catch really weird case where the persons state lookup exists but
+                        // but their office person record doesn't
+                        if (!doc.exists) {
+                            console.log(
+                                `found state legislator by name: ${newOfficePerson.displayName},
+                                but failed to find their 'office_people' record`
+                            );
+                        } else {
+                            // Handle merge
+                            newOfficePerson.mergeExistingOpenStatesData(doc.data(), person);
+
+                            // Create entirely new office person
+                            // This also updates the state legislator lookup table
+                            newOfficePerson.createNewStateLawMaker();
+
+                            // Delete old office people document
+                            firebase.firestore.collection('office_people').doc(checkResult.id).delete();
+
```

Comment:
i would factor this out into a delete function. keep all the firebase calls in one place